File: src/main/java/com/wafflestudio/draft/security/password/UserPrincipalDetailService.java
package com.wafflestudio.draft.security.password;

import com.wafflestudio.draft.model.User;
import com.wafflestudio.draft.repository.UserRepository;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class UserPrincipalDetailService implements UserDetailsService {
    private UserRepository userRepository;

    public UserPrincipalDetailService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }


    @Override
    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
        User user = this.userRepository.findByEmail(s)
                .orElseThrow(() -> new UsernameNotFoundException("User with email '%s' not found"));

        return new UserPrincipal(user);
    }


}

File: src/main/java/com/wafflestudio/draft/security/password/UserPrincipalDetailService.kt
package com.wafflestudio.draft.security.password

import com.wafflestudio.draft.repository.UserRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserPrincipalDetailService(private val userRepository: UserRepository) : UserDetailsService {
    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(s: String): UserDetails {
        val user = userRepository.findByEmail(s)
                .orElseThrow { UsernameNotFoundException("User with email '%s' not found") }
        return UserPrincipal(user)
    }

}